### PR TITLE
feat: Add XDG Base Directory specification support

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description

This PR implements XDG Base Directory specification support to enable better filesystem organization and package manager installations (Homebrew, Chocolatey, Winget).

- Link to the related issue(s): Issue #698
- Motivation: Current portable-only design prevents package manager installations and proper multi-user support. This implementation maintains 100% backward compatibility while enabling system-wide installations.

## 📋 Changes Summary

**Core implementation:**
- Added xdg_paths.py module for cross-platform path resolution via platformdirs
- Implemented dual installation mode detection (portable vs system-wide)
- Added first-run interactive prompt for mode selection (with non-interactive fallback)
- Deferred installation mode detection until after CLI argument parsing
- Ensured UpdateChecker and AdExtractor receive the detected installation mode

**Path organization:**

**Portable mode (existing installations):**
- Config: ./config.yaml
- Ads: ./**/*ad_*.{yaml,json}
- Downloaded ads: ./downloaded-ads/
- Browser profile: ./.temp/browser-profile
- Logs: ./kleinanzeigen_bot.log
- Update state: ./.temp/update_check_state.json

**System-wide mode (new XDG installations):**
- Config: $XDG_CONFIG_HOME/kleinanzeigen-bot/config.yaml
- Ads: $XDG_CONFIG_HOME/kleinanzeigen-bot/**/*ad_*.{yaml,json}
- Downloaded ads: $XDG_CONFIG_HOME/kleinanzeigen-bot/downloaded-ads/
- Browser profile: $XDG_CACHE_HOME/kleinanzeigen-bot/browser-profile
- Logs: $XDG_STATE_HOME/kleinanzeigen-bot/kleinanzeigen_bot.log
- Update state: $XDG_STATE_HOME/kleinanzeigen-bot/update_check_state.json

**Key features:**
- Automatic mode detection based on existing config file location
- User prompt on first run (interactive) with portable fallback (non-interactive)
- CLI overrides (--config, --logfile) take precedence over auto-detection
- Downloaded ads and update state use mode-aware paths

**Docs:**
- README now summarizes pros/cons of each installation mode with brief OS notes

**Testing:**
- Added/updated unit tests for installation mode wiring, XDG behavior, and browser lifecycle
- Translation coverage updated for new messages

### ⚙️ Type of Change
- [x] ✨ New feature (adds new functionality without breaking existing usage)
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 💥 Breaking change (changes that might break existing user setups, scripts, or configurations)

## ✅ Checklist
- [x] I have reviewed my changes to ensure they meet the project's standards.
- [x] I have tested my changes and ensured that all tests pass (`pdm run test`).
- [x] I have formatted the code (`pdm run format`).
- [x] I have verified that linting passes (`pdm run lint`).
- [x] I have updated documentation where necessary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

## 🔍 Technical Details

**Backward compatibility:**
- Existing ./config.yaml installations continue working unchanged
- No automatic migration - users must explicitly choose to move config
- All existing CLI flags and behaviors preserved

**Platform support:**
- Linux: ~/.config, ~/.cache, ~/.local/state
- macOS: ~/Library/Application Support, ~/Library/Caches, ~/Library/Logs
- Windows: %APPDATA%, %LOCALAPPDATA%

**Edge cases handled:**
- Unicode paths (NFD/NFC normalization) - verified with tests
- Non-interactive environments (CI/CD, scripts) - defaults to portable
- Custom config paths via --config - properly detected and respected
- Browser profile overrides via config.browser.user_data_dir

**Next steps (future PRs):**
- Package manager installation scripts (PyPi, Homebrew formula, Chocolatey package) #699 #762
- Migration guide documentation
- Optional migration helper command

## Summary by CodeRabbit

* **New Features**
  * Portable vs system-wide installation modes with auto-detection, first-run prompt, and mode-aware locations for config, logs, browser profile, downloads and update checks.

* **Bug Fixes**
  * Safer browser startup/profile handling, automatic cleanup on profile conflicts, clearer messages; update checks and downloads now respect installation mode.

* **Tests**
  * Extensive unit and integration coverage for path resolution, prompts, extraction/download flows and Unicode/NFC scenarios.

* **Documentation**
  * Added Installation modes section to README.

* **Chores**
  * Moved a utility dependency from dev to runtime.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## 📝 Review Notes

- DEBUG log messages are not translated (TranslatingLogger skips DEBUG), so debug strings were de-localized for consistency.
- When --remote-debugging-port is set, auto-defaulting user_data_dir and Chrome 136+ validation are skipped to avoid interfering with existing remote-debugging sessions.

### CODE REVIEW

File: src/kleinanzeigen_bot/utils/xdg_paths.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    xdg_config = get_xdg_base_dir("config") / "config.yaml"
+    LOG.debug(_("Checking for XDG config at: %s"), xdg_config)
+
+    if xdg_config.exists():
+        LOG.info(_("Detected installation mode: %s"), "xdg")
+        return "xdg"
+
+    # Neither exists - first run
+    LOG.info(_("No existing installation found"))
+    return None
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/utils/xdg_paths.py:105]: I like the translated DEBUG messages - users will see them using `-v`.

File: tests/unit/test_web_scraping_mixin.py [hunk 2]
```diff
@@ -1265,8 +1357,9 @@ async def mock_create_session(self:WebScrapingMixin) -> None:
 
         with patch("psutil.Process") as mock_proc:
             mock_proc.side_effect = psutil.NoSuchProcess(12345)
-            with pytest.raises(psutil.NoSuchProcess):
-                scraper.close_browser_session()
+            scraper.close_browser_session()
+            mock_browser.stop.assert_called_once()
+            assert scraper.browser is None
```
[the remark this answers is not in the document]
Comment (1cu) [reply tests/unit/test_web_scraping_mixin.py:1395]: Won't fix - out of scope.

File: src/kleinanzeigen_bot/utils/web_scraping_mixin.py [hunk 3]
```diff
@@ -501,16 +517,50 @@ def diagnose_browser_issues(self) -> None:
...
+                            p.kill()
+                        except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess, ValueError) as exc:
+                            LOG.debug("Failed to kill browser child process %s: %s", p, exc)
+                            # Ignore failures to kill already-exited or protected processes.
+                except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess, ValueError):
+                    continue
+            except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess, ValueError):
+                continue
+        self.page = None  # pyright: ignore[reportAttributeAccessIssue]
+        self.browser = None  # pyright: ignore[reportAttributeAccessIssue]
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/utils/web_scraping_mixin.py]: Won't fix - browser cleanup is out of scope for this PR.

File: src/kleinanzeigen_bot/utils/web_scraping_mixin.py [hunk 4]
```diff
@@ -40,6 +40,18 @@
...
+def _resolve_user_data_dir_paths(arg_value:str, config_value:str) -> tuple[Path | None, Path | None]:
+    """Resolve the argument and config user_data_dir paths for comparison."""
+    try:
+        return (
+            Path(arg_value).expanduser().resolve(),
+            Path(config_value).expanduser().resolve(),
+        )
+    except OSError as exc:
+        LOG.debug("Failed to resolve user_data_dir paths for comparison: %s", exc)
+        return None, None
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/utils/web_scraping_mixin.py:52]: No translation of DEBUG

**Comment by mrueg** [comment]
@1cu may I ask why you closed the PR?

**Comment by 1cu** [comment]
@mrueg Yes, but that will take a bit of time.
